library/general/src/lib/ui/event_dispatcher.rb

```diff
@@ -0,0 +1,65 @@
+require "yast"
+
+module UI
+  # Provides switch between event_loop and dispatching to handlers.
+  # @example simple OK/cancel dialog
+  #   class OKDialog
+  #     include Yast::UIShortcuts
+  #     include Yast::Logger
+  #     include UI::EventDispatcher
+  #     Yast.import "UI"
+  #
+  #     def run
+  #       return nil unless Yast::UI.OpenDialog(
+  #         HBox(
+  #           PushButton(Id(:ok), "OK"),
+  #           PushButton(Id(:cancel), "Cancel")
+  #         )
+  #       )
+  #       begin
+  #         return event_loop
+  #       ensure
+  #          Yast::UI.CloseDialog
+  #       end
+  #     end
+  #
+  #     def ok_handler
+  #       finish_dialog(:ok)
+  #       log.info "OK button pressed"
+  #     end
+  #   end
+  module EventDispatcher
+    # @internal constant to continue with dispatching
+    CONTINUE_WITH_DISPATCHING = :_dispatcher_continue
+
+    # Does UI event dispatching.
+    # @return value from exit_dialog method.
+    def event_loop
+      Yast.import "UI"
+      @_finish_dialog_dispatcher = CONTINUE_WITH_DISPATCHING
+
```

Comment:
Wouldn't it be more clear to use two instance variables? One boolean to decide when to break (let's say ```@_return_from_loop``` or ```@_continue_loop```) and another one to store the value to return (let's say ```@_return_value```). Not only the code would be cleaner, it would also save us from defining esoteric reserved symbols like ```:_dispatcher_continue```.